File: TelerikUIExamples/TelerikUIExamples/CalendarEventKitDataBinding.m
//
//  CalendarEventKitDataBinding.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "CalendarEventKitDataBinding.h"
#import <TelerikUI/TelerikUI.h>

@interface CalendarEventKitDataBinding ()

@property (nonatomic, strong) TKCalendar *calendarView;
@property (nonatomic, strong) TKCalendarEventKitDataSource *dataSource;

@end

@implementation CalendarEventKitDataBinding

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.dataSource = [TKCalendarEventKitDataSource new];
    
    self.calendarView = [[TKCalendar alloc] initWithFrame:self.exampleBounds];
    self.calendarView.dataSource = self.dataSource;
    [self.view addSubview:self.calendarView];
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    self.calendarView.frame = self.exampleBounds;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CalendarEventKitDataBinding.swift
//
//  CalendarEventKitDataBinding.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CalendarEventKitDataBinding: TKExamplesExampleViewController {
    
    let calendarView = TKCalendar()
    var dataSource = TKCalendarEventKitDataSource()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.dataSource = TKCalendarEventKitDataSource()
// >> populating-with-data-kitdatasource-swift
        self.calendarView.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.calendarView.dataSource = self.dataSource
        self.view.addSubview(self.calendarView)
// << populating-with-data-kitdatasource-swift
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        self.calendarView.frame = self.view.bounds
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
